src/node.cc

```diff
@@ -1467,12 +1467,12 @@ inline int Start(uv_loop_t* event_loop,
   if (isolate == nullptr)
     return 12;  // Signal internal error.
 
-  if (per_process::cli_options->print_version) {
+  if (per_process::cli_options->per_isolate->per_env->print_version) {
     printf("%s\n", NODE_VERSION);
     return 0;
   }
 
-  if (per_process::cli_options->print_v8_help) {
+  if (per_process::cli_options->per_isolate->per_env->print_v8_help) {
```

Comment:
Sorry – I wouldn’t move these, because they are handled during process startup, and the V8 options list in particular always hard-exits the process?

Reply:
Should we classify the options based on the states they affect (conceptually), or based on how they are currently handled? Out of the execution mode arguments:

- `syntax_check_only`
- `print_eval`
- `force_repl`
- `print_bash_completion`
- `print_help`
- `print_v8_help`
- `print_version`
- `prof_process`

I'd say at least `print_bash_completion` and `print_help` can also be handled during process startup, prior to Node's boostrapping - it just happens that they are implemented in JS at the moment. But these do not affect per-process states either.

Reply:
I would do the former, i.e. based on which states they affect – and at the same time, try to aim for narrower scopes (e.g. prefer per-env options over per-process).

We could definitely handle `print_bash_completion` and `print_help` during start-up, true … I don’t have strong feelings. If you want, you can move them to per-process options.

The prof-process thing is a bit different, imo, because you might be able to actually make use of its functionality when spawning e.g. a Worker or similar.

Reply:
> The prof-process thing is a bit different, imo, because you might be able to actually make use of its functionality when spawning e.g. a Worker or similar.

But should that functionality be available through CLI falgs? It feels quite odd to do `new Worker('???', { execArgv: [ '--prof-process', 'path/to/file' ] })`. Workers do not take `--eval` arguments that way either.

Reply:
> It feels quite odd to do `new Worker('???', { execArgv: [ '--prof-process', 'path/to/file' ] })`. 

I agree that it’s odd – and not a lot of people will have the idea to use this – but it matches the main script’s behaviour, so I think it’s okay?

> Workers do not take `--eval` arguments that way either.

I think that’s mostly a convenience thing :)